### ISSUE TITLE
Importing md file

### ISSUE THREAD

**Issue by fatihsezgin** [katherinepeterson/readme.so#142]
It would be very useful if we can upload a md file that is also has been created in the readme.so.

**Comment by jose-vale** [comment]
This is a pretty interesting feature, but I believe the implementation needs to be carefully considered.
How would we recognise the current sections from the imported README, compare titles or add metadata comments to the sections? What if the title was changed, we would need to treat it as a custom section, therefore this feature would always require the work in #6 to be completed first.
If we ended up adding some sort of comments to the sections to identify later, this would only be present in new README files, imports of older files would generate only custom sections.
@octokatherine what is your opinion regarding this?

**Comment by octokatherine** [comment]
I think a good path forward would be to complete #6 first for custom sections, then parse the readmes for titles. 

We should make it clear that it is intended to be used for md files that were created with readme.so, or at least formatted the same (h2's for headers) 

When we implement this, it would be great if the user could either upload the file or paste the file contents in. I believe more people use the copy functionality than the download functionality currently.

**Comment by jose-vale** [comment]
I can pick this up once #6 is closed, if that is okay with you. We can always discuss the smaller details when development starts.

**Comment by octokatherine** [comment]
Sounds good to me

**Comment by octokatherine** [comment]
@jose-vale are you still interested in working on this?

**Comment by jose-vale** [comment]
Yes, although I will only be able to pick it up next week. I was waiting for the custom sections development, will that need to be implemented in the context of this feature?

**Comment by octokatherine** [comment]
Custom sections has been implemented!

**Comment by jose-vale** [comment]
Ok, must have missed that update. Will start this as soon as I can.

**Comment by jose-vale** [comment]
Hey @octokatherine, I'm currently working on this. Do you have an icon for the import button? Also, I will only add translations to PT and EN, the rest will need to be done either at a later date, or by other contributors once I open the PR. Is that okay?

**Comment by MicahBCode** [comment]
@jose-vale Are you still working on this?

**Comment by jose-vale** [comment]
> @jose-vale Are you still working on this?

I started working on it, but have not had the time lately. Do you want to pick it up?

**Comment by MicahBCode** [comment]
> > @jose-vale Are you still working on this?
> 
> I started working on it, but have not had the time lately. Do you want to pick it up?

Yeah I might work with you on it together.
Maybe you can commit your changes you made already to a new branch in your forked github repo and add me as a contributor?

**Comment by jose-vale** [comment]
@MicahBCode I've added you to the repo. I don't recall how far along the feature was, but if you need anything, let me know.

**Comment by paoloricciuti** [comment]
In a somewhat related note: a cool and pretty rapid addition could be the one to export the metadata present in the local storage as json to let us export and import our work. Let me know if you are interested I can either open a new issue and I might also be able to work on this.

**Comment by BrookJeynes** [comment]
Hey @fatihsezgin @jose-vale @octokatherine @MicahBCode 

Based on issue #264 I've gone ahead and implemented something very similar in this PR #274

**Comment by Danny-dK** [comment]
@BrookJeynes The preview of this import and export looks and works great https://readme-so-git-fork-brookjeynes-264-impo-05d353-katherineoelsner.vercel.app/editor  Hope it can be merged soon.

**Comment by netopsengineer** [comment]
Was this feature abandoned @BrookJeynes your preview looked fantastic, especially for self hosting where you don't want to lose custom sections or have a template in mind you want to use.

**Comment by BrookJeynes** [comment]
CC: @Danny-dK @netopsengineer 

Sorry about that, here's the latest update https://github.com/octokatherine/readme.so/pull/288

**Comment by netopsengineer** [comment]
Thank you @BrookJeynes really hope this comes to fruition, great work!
